### PR TITLE
Restrict upload size for packages

### PR DESCRIPTION
**Pull request by shishirx34**
Fixes https://github.com/NuGet/NuGetGallery/issues/4823. This should also prevent the index lag caused by humongous file sizes, like the issue that happened today.

### CODE REVIEW

File: src/NuGetGallery/Strings.resx [hunk 1]
```diff
@@ -600,4 +600,7 @@ For more information, please contact '{2}'.</value>
   <data name="CannotEnqueueDueToReadOnly" xml:space="preserve">
     <value>Enqueuing unavailable: the gallery is currently in read only mode, with limited service. Please try again later.</value>
   </data>
+  <data name="UploadFileSizeExceedsMaxLimit" xml:space="preserve">
+    <value>The uploaded file is too big. Maximum allowed size is {0}MB for the upload file.</value>
```
Comment (shishirx34) [inline src/NuGetGallery/Strings.resx]: Feel free to suggest better wording here.

**Comment by scottbommarito** [comment]
I would make the max file size configurable, so that people hosting the gallery can set it as they want.

Otherwise, PR looks great
________________________________

**Comment by joelverhagen** [comment]
How many packages do we have today that violate this limit?

I thought we had this validation already. Did we lose it at some point?

File: tests/NuGetGallery.Facts/Controllers/PackagesControllerFacts.cs [hunk 2]
```diff
@@ -1905,6 +1905,27 @@ public class TheUploadFileActionForPostRequests
...
+                        GetConfigurationService(),
+                        uploadFileService: fakeUploadFileService);
+                    controller.SetCurrentUser(TestUtility.FakeUser);
+
+                    var postedFileBase = new Mock<HttpPostedFileBase>();
+                    postedFileBase.Setup(x => x.ContentLength).Returns(251 * 1024 * 1024 /*251MB*/);
+                    var result = await controller.UploadPackage(postedFileBase.Object) as JsonResult;
+
+                    Assert.NotNull(result);
+                    Assert.False(controller.ModelState.IsValid);
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/Controllers/PackagesControllerFacts.cs]: Assert message here to make sure the right validation is failing.

**Comment by joelverhagen** [comment]
How did we come up with the value 250 MB?

**Comment by shishirx34** [comment]
Svetlana pointed out that there is an IIS setting that should restrict big file uploads. I am testing this out.
The size value was determined by the documentation [here](https://docs.microsoft.com/en-us/nuget/policies/nuget-faq).

File: tests/NuGetGallery.Facts/Controllers/PackagesControllerFacts.cs (hunk 2 above)
Comment (shishirx34) [reply tests/NuGetGallery.Facts/Controllers/PackagesControllerFacts.cs]: sure.

File: src/NuGetGallery/Controllers/PackagesController.cs [hunk 3]
```diff
@@ -234,6 +234,11 @@ public virtual JsonResult UploadPackageProgress()
         public virtual async Task<JsonResult> UploadPackage(HttpPostedFileBase uploadFile)
         {
             var currentUser = GetCurrentUser();
+            if (uploadFile != null && uploadFile.ContentLength > Constants.MaxUploadFileSizeInBytes)
```
Comment (chenriksson) [inline src/NuGetGallery/Controllers/PackagesController.cs]: Can we move the 'if (uploadFile == null)' check above this, so we don't need another null check?

File: src/NuGetGallery/Strings.resx (hunk 1 above)
Comment (chenriksson) [reply src/NuGetGallery/Strings.resx]: How about just "The uploaded file exceeded the package size limit of 250MB."

**Comment by shishirx34** [comment]
Setting this via ASP.NETs property instead of code enforcer.

**Pull request closed without merging**